src/Microsoft.Health.Fhir.Core/Features/Search/SearchIndexEntry.cs

```diff
@@ -18,12 +18,26 @@ public class SearchIndexEntry : IEquatable<SearchIndexEntry>
         /// <summary>
         /// Initializes a new instance of the <see cref="SearchIndexEntry"/> class.
         /// </summary>
+        /// <remarks>
+        /// searchParameter and value parameters should not have a null value
+        /// But if the input json is not in the correct format then we are parsing the body here <see cref="FhirJsonInputFormatter"/> and passing the initial validations for required fields here <see cref="ModelAttributeValidator"/>
+        /// e.g. If the body contains Coverage.status = "", then after parsing Coverage.status = null & Coverage.statusElement = null, resulting into minimum cardinality error as expected
+        /// If the body contains Coverage.status = , then after parsing Coverage.status = null & Coverage.statusElement = {value=null}, which passes the Firely validation and CodeToTokenSearchValueConverter returns null
+        /// In this case return BadRequestException with a valid message instead of 500
+        /// </remarks>
         /// <param name="searchParameter">The search parameter</param>
         /// <param name="value">The searchable value.</param>
         public SearchIndexEntry(SearchParameterInfo searchParameter, ISearchValue value)
         {
-            EnsureArg.IsNotNull(searchParameter, nameof(searchParameter));
-            EnsureArg.IsNotNull(value, nameof(value));
+            if (searchParameter is null)
+            {
+                throw new BadRequestException(string.Format(Resources.ValueCannotBeNull, "Search Parameter"));
+            }
+
+            if (value is null)
+            {
+                throw new BadRequestException(string.Format(Resources.ValueCannotBeNull, searchParameter.Expression));
+            }
```

Comment:
Should the search indexer handle this instead? This still looks like an ArgumentException in this class.

```
  foreach (ISearchValue searchValue in ExtractSearchValues(
      searchParameter.Url.ToString(),
      searchParameter.Type,
      searchParameter.TargetResourceTypes,
      resource,
      searchParameter.Expression,
      context))
  {
     if(searchValue != null) {
      yield return new SearchIndexEntry(searchParameterInfo, searchValue);
     }
  }
```

or

```
  foreach (ISearchValue searchValue in ExtractSearchValues(
      searchParameter.Url.ToString(),
      searchParameter.Type,
      searchParameter.TargetResourceTypes,
      resource,
      searchParameter.Expression,
      context))
  {
     try {
      yield return new SearchIndexEntry(searchParameterInfo, searchValue);
     } catch(ArgumentException) { throw new BadRequestException(...) }
  }
```

Reply:
Added the null check in TypedElementSearchIndexer.ProcessNonCompositeSearchParameter. I noticed there is another function ProcessCompositeSearchParameter which calls the SearchIndexEntry too. Wondering if the null check would also apply to the composite Search parameter. Any thoughts?